marlowe-integration-tests/test/Language/Marlowe/Runtime/CliSpec.hs

```diff
@@ -0,0 +1,116 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Language.Marlowe.Runtime.CliSpec
+  where
+
+import Cardano.Api (AsType(AsTxBody), BabbageEra, CardanoEra(BabbageEra), TxBody, readFileTextEnvelope)
+import Control.Concurrent.Async.Lifted (concurrently)
+import qualified Control.Monad.Base as Trans
+import qualified Control.Monad.Reader as Reader
+import qualified Data.Aeson as Aeson
+import qualified Data.Either as Either
+import qualified Data.Maybe as Maybe
+import qualified Data.Set as Set
+import qualified Data.Text as Text
+import Data.Void (Void)
+import qualified Language.Marlowe.Core.V1.Semantics.Types as V1
+import Language.Marlowe.Runtime.Cardano.Api (cardanoEraToAsType)
+import Language.Marlowe.Runtime.ChainSync.Api (Address(..), Lovelace(Lovelace), TransactionMetadata, toBech32)
+import Language.Marlowe.Runtime.Client (runMarloweTxClient)
+import Language.Marlowe.Runtime.Core.Api (MarloweVersion(MarloweV1), MarloweVersionTag(V1), contractToJSON)
+import Language.Marlowe.Runtime.Integration.Common (Integration, Wallet(..), getGenesisWallet, runIntegrationTest)
+import Language.Marlowe.Runtime.Transaction.Api
+  (ContractCreated(..), CreateError, MarloweTxCommand(..), RoleTokensConfig(RoleTokensNone), WalletAddresses(..))
+import qualified Network.Protocol.Job.Client as JobClient
+import Test.Hspec (Spec, describe, it, shouldBe)
+import Test.Integration.Marlowe.Local
+  ( LocalTestnet(..)
+  , Workspace(Workspace, workspaceDir)
+  , execMarlowe_
+  , marloweSyncPort
+  , testnet
+  , withLocalMarloweRuntime
+  )
+
+serializeAddress :: Address -> String
+serializeAddress = Text.unpack . Maybe.fromJust . toBech32
+
+toCliArgs :: MarloweTxCommand Void (CreateError v) (ContractCreated BabbageEra v) -> [String]
+toCliArgs (Create _ _ WalletAddresses {changeAddress, extraAddresses} _ _ (Lovelace minUTXO) _) =
+  ["create", "--change-address", serializeAddress changeAddress]
+    <> do address <- Set.toList extraAddresses; ["--address", serializeAddress address]
+    <> ["--min-utxo", show minUTXO]
+
+marloweRuntimeCli :: [String] -> Integration ()
+marloweRuntimeCli cliArgs = do
+  marlowe_sync_port :: Int <- Reader.asks marloweSyncPort
```

Comment:
This is the reason you're seeing the handshake error - we need to connect to `marlowe-proxy`, not `marlowe-sync`.